testing/infra/terraform/modules/soaktest_workers/main.tf

```diff
@@ -0,0 +1,141 @@
+locals {
+  ssh_user_name      = "apmsoak"
+  executable_name    = "apmsoak"
+  service_name       = "apmsoak.service"
+  network_name       = "apmsoak-network-${terraform.workspace}"
+  remote_working_dir = "apmsoak/"
+  config_hash = sha256(join("_",
+    [
+      terraform.workspace,
+      var.apm_server_url,
+      var.apm_secret_token,
+      var.apm_loadgen_max_rate,
+      var.apm_loadgen_agents_replicas,
+      var.elastic_agent_version,
+      var.fleet_url,
+    ]
+  ))
+}
+
+resource "tls_private_key" "worker_login" {
+  algorithm = "RSA"
+  rsa_bits  = "4096"
+}
+
+data "tls_public_key" "worker_login" {
+  private_key_openssh = tls_private_key.worker_login.private_key_openssh
+}
+
+resource "google_compute_network" "worker" {
+  name = local.network_name
+}
+
+resource "google_service_account" "worker" {
+  account_id   = "apmsoak-${terraform.workspace}"
+  display_name = "APM Server soaktest worker"
+}
+
+resource "google_compute_firewall" "allow_ssh" {
+  name    = "apmsoak-allowssh-${terraform.workspace}"
+  network = google_compute_network.worker.self_link
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+  source_ranges = ["0.0.0.0/0"]
+}
+
+data "google_compute_image" "worker_image" {
+  project = "ubuntu-os-cloud"
+  family  = "ubuntu-minimal-2004-lts"
+}
+
+resource "google_compute_instance" "worker" {
+  # Trigger a recreate of workers on config changes
+  name         = substr("apmsoak-${local.config_hash}", 0, 63)
+  machine_type = "e2-micro"
+  zone         = var.gcp_zone
+
+  boot_disk {
+    initialize_params {
+      image = data.google_compute_image.worker_image.self_link
+    }
+  }
+
+  network_interface {
+    network = local.network_name
```

Comment:
shouldn't we be using `google_compute_network.worker.self_link` here?

Reply:
Good catch, name and self_link both works but self_link will be better. Will update this.